### ISSUE TITLE
Feature suggestion?: inverse search for labels when cursor is over \cref{}

### ISSUE THREAD

**Issue by kathode** [LaTeXing/LaTeXing#136]
Hi there,

I have been trying to get LaTeXing/Sublime to show me the definition of a particular label X when the cursor is over a particular \cref{X,Y,Z}. This feature is available for \cite{} commands, and would be very useful to have a similar feature for \cref{} or \ref{} commands, particularly when dealing with very large projects (which is my case).

I have tried using F12, as suggested online, to lookup the definition of a label without success. For example, in the following screenshot, I am trying to lookup the definition of 'lemma:PBMH:alternative-1' when the mouse cursor is over \cref{lemma:PBMH:alternative-1}. However, Sublime notes on the status bar at the bottom that it is unable to find it.

![unable-to-find-lemma-pbmh](https://cloud.githubusercontent.com/assets/5167368/7241926/ecb82c0a-e7b5-11e4-8e7a-2e9be14037eb.png)

Admitting that it wants the whole key to be considered instead, I highlight it, and press F12. However, Sublime is still unable to find the definition as shown in the following screenshot.

![unable-to-find-lemma-alternative-1](https://cloud.githubusercontent.com/assets/5167368/7241831/d3dfa9f2-e7b4-11e4-97da-f66fa6a3e713.png)

If I use Goto > Goto symbol in project, then I can manually type the key, and subsequently by choosing the right occurrence I can be shown the definition at the right location, as shown in the following screenshot.

![screen shot 2015-04-20 at 23 20 45](https://cloud.githubusercontent.com/assets/5167368/7241962/43ee36b8-e7b6-11e4-9b80-06d8a976e8a1.png)

I am unsure as to whether this is supposed to be provided by Sublime or LaTeXing, or whether none of this is intended. However, my previous LaTeX editor used to be able to show me the definition of a particular item based on its label, similar to how editors deal with code, where it is straightforward to lookup the definition of a particular function. (In addition, it could also flag whether a key being used in a \cite or \ref command exists at all, and also whether duplicates have been specified, but that would be another feature in itself).

I realise I may be asking too much, but this would certainly be a very useful feature for my usage scenario.

Thanks

**Comment by csch0** [comment]
Hi @kathode,

you are talking about the goto function provided by Sublime Text is limited to syntax definitions, its not implemented by LaTeXing. LaTeXing has its own key combination http://docs.latexing.com/stable/user-guide.html#open-command for that which is not just working for cite or ref's

**Comment by kathode** [comment]
Hi @csch0,

Apologies, I didn't realise the combination is CMD+L followed by CMD+K. I had tried a couple of them before, obviously with no success. Being a newcomer to Sublime, it was not clear that it implied a two-step key combination.

From my testing, it works reasonably well. However, when I consider a \cref{X,Y} command involving more than one cross reference, it seems to take the whole string "X,Y" for which no label is defined, rather than my current selection 'X' or 'Y', or just the label under the cursor. It would be nice if that could be fixed.

Furthermore, it also does not seem to work with any labels involving parentheses. I wonder if that is related to the previous issue I raised.

Thanks a lot for the quick response!
